lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -37,6 +44,17 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
+          if warning_about_bare_error && !negative_expectation
+            RSpec.warning("Using the `raise_error` matcher without providing a specific " \
+                          "error or message risks false positives, since `raise_error` " \
+                          "will match when Ruby raises a `NoMethodError`, `NameError` or " \
+                          "`ArgumentError`, potentially allowing the expectation to pass " \
+                          "without even executing the method you are intending to call. " \
+                          "Instead consider providing a specific error class or message. " \
+                          "This message can be supressed by setting: " \
+                          "`RSpec::Expectations.configuration.warn_about_false_positives = false`")
+          end
```

Comment:
I think these 10 lines could benefit from being extracted into a helper method.  It's a lot to have in-lined into `matches?`.